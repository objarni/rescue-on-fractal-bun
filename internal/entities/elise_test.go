package entities

import (
	"fmt"
	approvals "github.com/approvals/go-approval-tests"
	px "github.com/faiface/pixel"
	"objarni/rescue-on-fractal-bun/internal/events"
	"objarni/rescue-on-fractal-bun/tests"
	"strings"
	"testing"
)

/*
- #initial state
- walking right
- walking left
- jumping when standing still
- jumping when pressing right
- clicking action when standing still
- clicking action when mid-air
- taking damage
*/

func Test_eliseInitial(t *testing.T) {
	elise := MakeElise(px.V(10, 20))
	var state = printElise(elise)
	approvals.VerifyString(t, state)
}

func Test_pressingLeft(t *testing.T) {
	elise := MakeElise(px.V(10, 20))
	elise = elise.Handle(EventBox{
		Event: events.KeyLeftDown,
		Box:   px.Rect{},
	})
	elise = elise.Tick(0, nil)
	var state = printElise(elise)
	approvals.VerifyString(t, state)
}

func printElise(elise Entity) string {
	state := fmt.Sprintf("Elise %v", "standing")
	hb := fmt.Sprintf("HitBox %v", printRect(elise.HitBox()))
	facing := fmt.Sprintf("Facing right")
	all := []string{state, hb, facing}
	return strings.Join(all, "\n") + "\n"
}

func printRect(box px.Rect) interface{} {
	return fmt.Sprintf("[%1.0f,%1.0f->%1.0f,%1.0f]",
		box.Min.X, box.Min.Y, box.Max.X, box.Max.Y)
}

func init() {
	approvals.UseReporter(tests.ReportWithMeld())
}
