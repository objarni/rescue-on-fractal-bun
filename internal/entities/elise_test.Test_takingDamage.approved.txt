*** Scenario ***
* Events:
Damage [-10,-10->10,10]

Ground height: 0
Tick count: 1

* Elise start state:
Elise Standing
Vel: <0.0,0.0>
Facing right
Gfx:
Moved 0 pixels right 50 pixels up:
  Image "IEliseWalk2"

* Elise end state:
Elise Standing
Vel: <0.0,-0.1>
Facing right
Gfx:
Moved 5 pixels right 50 pixels up:
  Image "IEliseWalk2"


*** Simulation ***

 * Tick 1 *
Entity 0 is at [-10,0->10,100]
Event: Wall [-1000,-1000->1000,0]
Event: Damage [-10,-10->10,10]
Consequences:
- Entity 0 needs to handle Damage

